.github/workflows/daily_property_tests.yaml

```diff
@@ -0,0 +1,71 @@
+name: Daily property test run
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: "0 12 * * *"
+  pull_request:
+    paths:
+      - ".github/workflows/daily_property_tests.yaml"
+
+permissions:
+  contents: read
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
+env:
+  CARGO_INCREMENTAL: 0
+  CARGO_NET_RETRY: 10
+  CARGO_TERM_COLOR: always
+  RUSTUP_MAX_RETRIES: 10
+  FORCE_COLOR: 1
+
+jobs:
+  property_tests:
+    name: Property tests
+    runs-on: ubuntu-latest
+    timeout-minutes: 20
+    # Don't run the cron job on forks:
+    if: ${{ github.repository == 'astral-sh/ruff' || github.event_name != 'schedule' }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          persist-credentials: false
+      - name: "Install Rust toolchain"
+        run: rustup show
+      - name: "Install mold"
+        uses: rui314/setup-mold@v1
+      - uses: Swatinem/rust-cache@v2
+      - name: Build Red Knot
+        # A debug build means the script runs slower once it gets started,
+        # but this is outweighed by the fact that a release build takes *much* longer to compile in CI
```

Comment:
My reading of this comment would suggest that the next line would do a debug build, but the line below appears to do a release build? (I personally don't have strong feelings either way, just curious if this comment should be updated to explain why you actually chose a release build.)

Reply:
Yes, thanks. Should have left this in draft mode until I gathered all the data. I ran two experiments.

Release mode: https://github.com/astral-sh/ruff/actions/runs/12678401447 (build: 1m 53s, five test iterations: 1m 26s)
Debug mode: https://github.com/astral-sh/ruff/actions/runs/12678489298/job/35336173933 (build: 53s, five test iterations: ~ 14 minutes, not finished yet)

The number of tests (5 iterations × 100000 tests / iteration = 0.5 million tests) is not chosen completely arbitrarily. From local tests, this seems to be the right order of magnitude to uncover some bugs that were not detected at one or two orders of magnitude fewer.

So release mode seems like the way to go. Will update that comment.